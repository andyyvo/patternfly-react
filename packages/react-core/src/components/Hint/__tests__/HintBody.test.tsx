import React from "react";
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { HintBody } from "../HintBody";

describe('HintBody', () => {
  it('renders without children', () => {
    render(<HintBody data-testid="HintBody"></HintBody>);
    
    expect(screen.getByTestId('HintBody')).toBeVisible();
  });

  it('renders children', () => {
    render(<HintBody>{<button>Test Me</button>}</HintBody>);

    expect(screen.getByRole('button', { name: 'Test Me' })).toBeVisible();
  });

  it('renders with class pf-c-hint__body', () => {
    render(<HintBody>Hint Body Test</HintBody>);

    const body = screen.getByText('Hint Body Test');

    expect(body).toHaveClass('pf-c-hint__body');
  });

  it('renders with custom class names provided via prop', () => {
    render(<HintBody className="custom-classname">Hint Body Test</HintBody>);

    const body = screen.getByText('Hint Body Test');

    expect(body).toHaveClass('custom-classname');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<HintBody>{<button>Test</button>}</HintBody>);

    expect(asFragment()).toMatchSnapshot();
  });
});