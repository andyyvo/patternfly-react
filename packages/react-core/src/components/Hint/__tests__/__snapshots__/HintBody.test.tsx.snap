// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HintBody matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pf-c-hint__body"
  >
    <button>
      Test
    </button>
  </div>
</DocumentFragment>
`;
